sospice/util/sigma_clipping.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from scipy.ndimage import generic_filter
+from numpy import ma
+
+
+def _stddev_loc(image, size):
+    if type(size) is int:
+        size = (size,) * image.ndim
+    mean2 = _blur(image, size) ** 2
+    vari = _blur(image ** 2, size)
+    vari -= mean2
+    vari[vari <= 0] = 1e-20
+    return np.sqrt(vari)
+
+
+def _blur(image, size, output=None):
+
+    if type(size) is int:
+        size = (size,)*image.ndim
+
+    if output is None:
+        output = np.empty_like(image)
+
+    if image.ndim == 2:
+        generic_filter(image, np.nanmean, mode='reflect', output=output, size=size)
+    elif image.ndim == 3:
```

Comment:
No need to distinguish cases when using `generic_filter()`? Actually `_blur()` could work for any number of dimensions.